stable_baselines3/common/base_class.py

```diff
@@ -513,14 +375,254 @@ def _update_info_buffer(self, infos: List[Dict[str, Any]], dones: Optional[np.nd
+        - action_space
+
+        :param env: The environment for learning a policy
+        """
+        check_for_correct_spaces(env, self.observation_space, self.action_space)
+        # it must be coherent now
+        # if it is not a VecEnv, make it a VecEnv
+        env = self._wrap_env(env)
+
+        self.n_envs = env.num_envs
+        self.env = env
+
+    @abstractmethod
+    def learn(
+        self,
+        total_timesteps: int,
+        callback: MaybeCallback = None,
+        log_interval: int = 100,
+        tb_log_name: str = "run",
+        eval_env: Optional[GymEnv] = None,
+        eval_freq: int = -1,
+        n_eval_episodes: int = 5,
+        eval_log_path: Optional[str] = None,
+        reset_num_timesteps: bool = True,
+    ) -> "BaseAlgorithm":
+        """
+        Return a trained model.
+
+        :param total_timesteps: (int) The total number of samples (env steps) to train on
+        :param callback: (MaybeCallback) callback(s) called at every step with state of the algorithm.
+        :param log_interval: (int) The number of timesteps before logging.
+        :param tb_log_name: (str) the name of the run for TensorBoard logging
+        :param eval_env: (gym.Env) Environment that will be used to evaluate the agent
+        :param eval_freq: (int) Evaluate the agent every ``eval_freq`` timesteps (this may vary a little)
+        :param n_eval_episodes: (int) Number of episode to evaluate the agent
+        :param eval_log_path: (Optional[str]) Path to a folder where the evaluations will be saved
+        :param reset_num_timesteps: (bool) whether or not to reset the current timestep number (used in logging)
+        :return: (BaseAlgorithm) the trained model
+        """
+
+    def predict(
+        self,
+        observation: np.ndarray,
+        state: Optional[np.ndarray] = None,
+        mask: Optional[np.ndarray] = None,
+        deterministic: bool = False,
+    ) -> Tuple[np.ndarray, Optional[np.ndarray]]:
+        """
+        Get the model's action(s) from an observation
+
+        :param observation: (np.ndarray) the input observation
+        :param state: (Optional[np.ndarray]) The last states (can be None, used in recurrent policies)
+        :param mask: (Optional[np.ndarray]) The last masks (can be None, used in recurrent policies)
+        :param deterministic: (bool) Whether or not to return deterministic actions.
+        :return: (Tuple[np.ndarray, Optional[np.ndarray]]) the model's action and the next state
+            (used in recurrent policies)
+        """
+        return self.policy.predict(observation, state, mask, deterministic)
+
+    def set_random_seed(self, seed: Optional[int] = None) -> None:
+        """
+        Set the seed of the pseudo-random generators
+        (python, numpy, pytorch, gym, action_space)
+
+        :param seed: (int)
+        """
+        if seed is None:
+            return
+        set_random_seed(seed, using_cuda=self.device == th.device("cuda"))
```

Comment:
since #154 I realized that we should check `self.device.type`, should we fix it here or in a separate PR?

Reply:
Sounds like something that could be included here, although would you be able to add it if it is a quick thing? I am reading through the PR but only slowly digesting how it works.

Also any ideas what could be causing the flake8 linting error exactly? My flake8 is not catching it :/